en/scripting/language-support.md

```diff
@@ -1,40 +1,108 @@
+
 # Language Support
 
 ## TypeScript
 
-Cocos Creator 3.0 uses [babel](https://babeljs.io/) instead of [tsc](https://www.typescriptlang.org/) to compile __TypeScript__ scripts. In particular, the [@babel/plugin-transform-typescript](https://babeljs.io/docs/en/babel-plugin-transform-typescript) plugin is used. For this reason, TypeScript support has certain limitations.
+Cocos Creator supports TypeScript 4.1.0. The following restrictions have been made on this base:
```

Comment:
what does: "made on this base" mean?

Reply:
It means the following restrictions are based on TypeScript 4.1.0.